x-pack/plugins/cases/server/client/attachments/bulk_delete.ts

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import Boom from '@hapi/boom';
+import { pipe } from 'fp-ts/lib/pipeable';
+import { fold } from 'fp-ts/lib/Either';
+import { identity } from 'fp-ts/lib/function';
+
+import pMap from 'p-map';
+import { partition } from 'lodash';
+import type { File } from '@kbn/files-plugin/common';
+import type { FileServiceStart } from '@kbn/files-plugin/server';
+import { BulkDeleteFileAttachmentsRequestRt, excess, throwErrors } from '../../../common/api';
+import { MAX_CONCURRENT_SEARCHES } from '../../../common/constants';
+import type { CasesClientArgs } from '../types';
+import { createCaseError } from '../../common/error';
+import type { OwnerEntity } from '../../authorization';
+import { Operations } from '../../authorization';
+import type { BulkDeleteFileArgs } from './types';
+import { constructOwnerFromFileKind, CaseFileMetadataForDeletionRt } from '../../../common/files';
+import type { CasesClient } from '../client';
+
+export const bulkDeleteFileAttachments = async (
+  { caseId, fileIds }: BulkDeleteFileArgs,
+  clientArgs: CasesClientArgs,
+  casesClient: CasesClient
+) => {
+  const {
+    user,
+    services: { attachmentService, userActionService },
+    logger,
+    authorization,
+    fileService,
+  } = clientArgs;
+
+  try {
+    const request = pipe(
+      excess(BulkDeleteFileAttachmentsRequestRt).decode({ ids: fileIds }),
+      fold(throwErrors(Boom.badRequest), identity)
+    );
+
+    await casesClient.cases.resolve({ id: caseId, includeComments: false });
+
+    const fileEntities = await getFileEntities(caseId, request.ids, fileService);
+
+    // It's possible for this to return an empty array if there was an error creating file attachments in which case the
+    // file would be present but the case attachment would not
+    const fileAttachments = await attachmentService.getter.getFileAttachments({
+      caseId,
+      fileIds: request.ids,
+    });
+
+    await authorization.ensureAuthorized({
+      entities: [
+        ...fileAttachments.map((attachment) => ({
+          id: attachment.id,
+          owner: attachment.attributes.owner,
+        })),
+        ...fileEntities,
```

Comment:
In addition to checking if the user can delete the `attachments` is this checking if he can also delete the files in the `fileService`?

And if so, is `operation: Operations.deleteComment` the correct one for that, isn't this `Cases` specific?